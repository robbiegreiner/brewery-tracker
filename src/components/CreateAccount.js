import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link, Redirect } from 'react-router-dom';
import firebase, { auth, provider } from '../firebase.js';

class CreateAccount extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: ''
    };
  }


  handleChange(description, event){
    this.setState({
      [description]: event.target.value
    });
  }

  createNewUser() {
    const { createAccount } = this.props;
    createAccount(this.state.email, this.state.password);
  }

  render() {
    return (
      <div className='create-account'>
        { this.props.user.id ? <Redirect to='/'/> : null}
        <h1>Create Account Here</h1>
        <input placeholder="email" onChange={(event) => this.handleChange('email', event) }></input>
        <input placeholder="password" onChange={(event) => this.handleChange('password', event) }></input>
        <button className='create-btn' onClick={ () => this.createNewUser()}>submit</button>
      </div>
    );

  }
}

CreateAccount.propTypes = {
  createAccount: PropTypes.func,
  user: PropTypes.object
};

export default CreateAccount;
