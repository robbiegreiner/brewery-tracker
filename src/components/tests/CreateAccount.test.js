import React from "react";
import CreateAccount from '../CreateAccount';
import { shallow } from "enzyme";

describe('CreateAccount component', () => {
  const mockFn = jest.fn();

  const wrapper = shallow(<CreateAccount
    user={{}}
    errorMessage={''}
    createAccount={mockFn}/>);


  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have an empty default state', () => {
    expect(wrapper.state().email).toEqual('');
    expect(wrapper.state().password).toEqual('');
  });

  it('should render two inputs', () => {
    const inputs = wrapper.find('input');

    expect(inputs.length).toEqual(2);
  });

  it('should change state on input', () => {
    const emailInput = wrapper.find('.email-input');
    const passwordInput = wrapper.find('.password-input');

    emailInput.simulate('change', { target: { value: 'robbie@gmail.com' } });
    passwordInput.simulate('change', { target: { value: 'password' } });
    expect(wrapper.state().email).toEqual('robbie@gmail.com');
    expect(wrapper.state().password).toEqual('password');
  });

  it('should run function on submit', () => {
    const button = wrapper.find('button').first();

    button.simulate('click');
    expect(mockFn).toHaveBeenCalledTimes(1);
  });


});
