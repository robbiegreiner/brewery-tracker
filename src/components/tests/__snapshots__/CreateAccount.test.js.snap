// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateAccount component should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateAccount
    createAccount={[Function]}
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <h1>
          Create Account Here
</h1>,
        <input
          onChange={[Function]}
          placeholder="email"
/>,
        <input
          onChange={[Function]}
          placeholder="password"
/>,
        <button
          className="create-btn"
          onClick={[Function]}
>
          submit
</button>,
      ],
      "className": "create-account",
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Create Account Here",
        },
        "ref": null,
        "rendered": "Create Account Here",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "email",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "password",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "submit",
          "className": "create-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "submit",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <h1>
            Create Account Here
</h1>,
          <input
            onChange={[Function]}
            placeholder="email"
/>,
          <input
            onChange={[Function]}
            placeholder="password"
/>,
          <button
            className="create-btn"
            onClick={[Function]}
>
            submit
</button>,
        ],
        "className": "create-account",
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Create Account Here",
          },
          "ref": null,
          "rendered": "Create Account Here",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "email",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "password",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "submit",
            "className": "create-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "submit",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
